cmd/ko/publish.go

```diff
@@ -15,22 +15,54 @@
 package main
 
 import (
+	"fmt"
+	gb "go/build"
 	"log"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 
 	"github.com/google/go-containerregistry/pkg/ko/build"
 	"github.com/google/go-containerregistry/pkg/ko/publish"
 	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/daemon"
 )
 
+func qualifyLocalImport(importpath, gopathsrc, pwd string) (string, error) {
+	if !strings.HasPrefix(pwd, gopathsrc) {
+		return "", fmt.Errorf("pwd (%q) must be on $GOPATH/src (%q) to support local imports", pwd, gopathsrc)
+	}
+	// Given $GOPATH/src and $PWD (which must be within $GOPATH/src), trim
+	// off $GOPATH/src/ from $PWD and append local importpath to get the
+	// fully-qualified importpath.
+	return filepath.Join(strings.TrimPrefix(pwd, gopathsrc+string(filepath.Separator)), importpath), nil
+}
+
 func publishImages(importpaths []string, lo *LocalOptions) {
 	b, err := build.NewGo(gobuildOptions())
 	if err != nil {
 		log.Fatalf("error creating go builder: %v", err)
 	}
 	for _, importpath := range importpaths {
+		if gb.IsLocalImport(importpath) {
+			// Interpret `ko publish ./cmd/foo` as `ko publish $GOPATH/cmd/foo`
```

Comment:
Is this comment correct? `$GOPATH/cmd/foo` seems unlikely.

Reply:
It was not. Updated.